laws/contravariant.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const {identity, compose} = require('fantasy-combinators');
+const {contramap} = require('..');
+
+/*
+
+### Contravariant Functor
+
+1. `u.contramap(a => a)` is equivalent to `u` (identity)
+2. `u.contramap(x => f(g(x)))` is equivalent to `u.contramap(f).contramap(g)` (composition)
+
+*/
+
+const identityʹ = t => eq => x => {
+    const a = t(x)[contramap](identity);
+    const b = t(x);
+    return eq(a, b);
+};
+
+const composition = t => eq => x => {
+    const a = t(x)[contramap](compose(identity)(identity));
```

Comment:
I'm wondering whether we should use some contravariant function here rather than identity, as we don't really ensure correct composition order while using `identity`.

Not sure what the best way to generate an arbitrary function from within these tests would be though, other than having some memoised function that returns a randomly generated number e.g.

```js
const fn = () => {
  const received = new Map()
  return x => {
    if (!received.has(x)) received.set(x, Math.random())
    return received.get(x)
  }
}

const composition = t => eq => x => {
    const f = fn();
    const g = fn();
    const a = t(x)[contramap](compose(f)(g));
    const b = t(x)[contramap](f)[contramap](g);
    return a === b;
};
```

I believe the same issue exists in the profunctor test too.